src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    public static final int AGE = 18;
+    public static final int PASSWORD = 6;
+    private RegistrationServiceImpl regService = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user;
+
+    @Test
+    void register_Login_NotOk() {
+        User testUser = new User();
+        user = new User();
+        testUser.setAge(20);
+        testUser.setLogin("Mersedes");
+        testUser.setPassword("king_1975");
+        storageDao.add(testUser);
+        user.setLogin("Mersedes");
+        assertThrows(RuntimeException.class, () ->
+                regService.register(user).equals(storageDao.get(user.getLogin())));
+    }
+
+    @Test
+    void register_passwordIsNull_NotOk() {
+        user = new User();
+        user.setAge(19);
+        user.setLogin("kijuhzt");
+        assertThrows(RuntimeException.class, () -> regService.register(user));
+    }
+
+    @Test
+    void register_loginIsNull_NotOk() {
+        user = new User();
+        user.setAge(19);
+        assertThrows(RuntimeException.class, () -> regService.register(user));
```

Comment:
```suggestion
        User user = new User();
        user.setPassword("somevalidpassword");
        user.setAge(19);
        assertThrows(RuntimeException.class, () -> regService.register(user));
```

Знову ж таки, якщо ми тестимо користувача з null логіном, всі інші поля в нього мають бути валідні. І краще кожен раз в кожному тесті створювати нового користувача, щоб дані з попереднього тесту ніяк не впливали на новий